v2/json2/server.go

```diff
@@ -185,8 +186,8 @@ func (c *CodecRequest) WriteError(w http.ResponseWriter, status int, err error)
 }
 
 func (c *CodecRequest) writeServerResponse(w http.ResponseWriter, res *serverResponse) {
-	// Id is null for notifications and they don't have a response.
-	if c.request.Id != nil {
+	// Id is null for notifications and they don't have a response. OR it just parse error
```

Comment:
`// Id is null for notifications and they don't have a response, or a parsing error has occurred`